idseq_pipeline/commands/accessionid2seq.py

```diff
@@ -9,9 +10,142 @@
+                    if entry:
+                        nt.seek(entry[0]+entry[1], 0)
+                        ref_seq = nt.read(entry[2])
+                        accession_dict['ref_seq'] = ref_seq.replace("\n", "")
+                    else:
+                        accession_dict['ref_seq'] = 'NOT FOUND'
+
+                result_dict[family_id][genus_id][species_id][accession_id] = accession_dict
+    print("%d lines in the m8 file" % line_count)
+
+    # output json by species, genus, family
+    execute_command("mkdir -p %s " % output_json_dir)
+    for (family_id, family_dict) in result_dict.iteritems():
+        with open("%s/%s.family.%d.align_viz.json" %(output_json_dir, db_type.lower(), int(family_id)), 'wb') as outjf:
+            json.dump(family_dict, outjf)
+        for (genus_id, genus_dict) in family_dict.iteritems():
+            with open("%s/%s.genus.%d.align_viz.json" %(output_json_dir, db_type.lower(), int(genus_id)), 'wb') as outjf:
+                json.dump(genus_dict, outjf)
+            for (species_id, species_dict) in genus_dict.iteritems():
+                with open("%s/%s.species.%d.align_viz.json" %(output_json_dir, db_type.lower(), int(species_id)), 'wb') as outjf:
+                    json.dump(species_dict, outjf)
+    return "Read2Seq Size: %d, M8 lines %d" % (len(read2seq),line_count)
 
 class Accessionid2seq(Base):
     def run(self):
-        print("Placeholder")
+        from .common import * #TO DO: clean up the imports across this package
+
+        # Make work directory
+        dest_dir = os.path.join(DEST_DIR, "accession2seq/tmp-%d" % os.getpid())
+        execute_command("mkdir -p %s" % dest_dir)
+        arguments = self.options
+        s3_db_path = arguments.get('--s3_db_path')
+        s3_db_loc_path = arguments.get('--s3_db_loc_path')
+        db_type = arguments.get('--db_type')
+        input_fasta_s3_path = arguments.get('--input_fasta_s3_path')
+        input_m8_s3_path = arguments.get('--input_m8_s3_path')
+        output_json_s3_path = arguments.get('--output_json_s3_path')
+        output_json_s3_path = arguments.get('--output_json_s3_path').rstrip('/')
+
+        local_db_path = arguments.get('--local_db_path')
+
+
+        local_db_loc_path =  os.path.join(dest_dir, os.path.basename(s3_db_loc_path))
+        local_fasta_path =  os.path.join(dest_dir, os.path.basename(input_fasta_s3_path))
+        local_m8_path = os.path.join(dest_dir, os.path.basename(input_m8_s3_path))
+        local_json_path = os.path.join(dest_dir, "align_viz")
+
+        # Copy the data over from s3
+        if not local_db_path:
+            local_db_path =  os.path.join(dest_dir, os.path.basename(s3_db_path))
+            if s3_db_path[-3:] == '.gz':
+                local_db_path = local_db_path[:-3]
+                execute_command("aws s3 cp %s - | gunzip > %s" %(s3_db_path, local_db_path))
+            else:
+                execute_command("aws s3 cp %s %s" %(s3_db_path, local_db_path))
```

Comment:
This entire block you can replace with
```
local_db_path  = fetch_from_s3(s3_db_path, dest_dir, auto_unzip=True)
```
leaving the rest of the code unchanged.   It will correctly skip the download if the file already exists.

Reply:
I won't use the run function here. I will just use the helper function for stage 2